### PR TITLE
[Feature] Select menu default values 

### PR DESCRIPTION
**Pull request by Misha-133**
This PR adds support for new default values property of a select menu & the new `Resolved` property to `IUserMessage`
[docs PR](https://github.com/discord/discord-api-docs/pull/6442)

### Changes
- add new prop to select menu, select menu builder, component builder
- add `Resolved` to `IUserMessage`

### CODE REVIEW

File: src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectDefaultValueType.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+namespace Discord;
+
+/// <summary>
+///     Type of a <see cref="SelectDefaultValueType" />.
+/// </summary>
+public enum SelectDefaultValueType
+{
+    User = 0,
```
Comment (quinchs) [inline src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectDefaultValueType.cs:11]: Docs?

File: src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectMenuDefaultValue.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+namespace Discord;
+
+/// <summary>
+///     Represents a default value of an auto-populated select menu.
+/// </summary>
+public struct SelectMenuDefaultValue
```
Comment (quinchs) [inline src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectMenuDefaultValue.cs]: could be made `readonly`

File: src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectMenuDefaultValue.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+{
+    public ulong Id { get; }
```
Comment (quinchs) [inline src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectMenuDefaultValue.cs:11]: can just be a readonly field

File: src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectMenuDefaultValue.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    public SelectMenuDefaultValue(ulong id, SelectDefaultValueType type)
+    {
+        Id = id;
+        Type = type;
+    }
+
+    /// <summary>
+    ///     Creates a new default value from a <see cref="IChannel"/>.
+    /// </summary>
+    public SelectMenuDefaultValue FromChannel(IChannel channel)
```
Comment (quinchs) [inline src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectMenuDefaultValue.cs]: Are these meant to be `static`?

File: src/Discord.Net.Rest/Entities/Messages/RestUserMessage.cs [hunk 5]
```diff
@@ -2,6 +2,8 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Diagnostics;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
```
Comment (quinchs) [inline src/Discord.Net.Rest/Entities/Messages/RestUserMessage.cs]: ?

**Review by quinchs — CHANGES_REQUESTED** [review]

File: src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectMenuDefaultValue.cs (hunk 2 above)
Comment (Misha-133) [reply src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectMenuDefaultValue.cs]: Hmm true. 
Made it `readonly`

File: src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectMenuDefaultValue.cs (hunk 3 above)
Comment (Misha-133) [reply src/Discord.Net.Core/Entities/Interactions/MessageComponents/SelectMenuDefaultValue.cs:11]: The `struct` itself is `readonly` now, so no need to mark the field as one

File: src/Discord.Net.Rest/Entities/Messages/RestUserMessage.cs (hunk 5 above)
Comment (Misha-133) [reply src/Discord.Net.Rest/Entities/Messages/RestUserMessage.cs]: VS moment

**Review by MrCakeSlayer — APPROVED** [review]

**Review by quinchs — APPROVED** [review]

**Pull request merged**